### DISCUSSION TITLE
Homebrew detecting auto-updating casks

### DISCUSSION THREAD

**Discussion by backstab5983** [Homebrew/discussions#4753] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: d4444b563e24ac7c05a93121c464c02dfa04d44f
Last commit: 4 days ago
Core tap JSON: 31 Aug 11:37 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Why can't Homebrew detect the version of auto updating apps originally installed with Homebrew? Is there any way to force the "reading" of the actual version of the app installed?

For example if I run `brew upgrade --greedy --verbose` it will show the original version of the cask when it was installed but it cannot detect that some of these apps (the auto updating ones) have been updated in app and it will show the wrong version installed and perhaps try to force the update.

Example: 
`==> Upgrading 5 outdated packages:
brave-browser 1.57.53.0,157.53 -> 1.57.57.0,157.57
discord 0.0.276 -> 0.0.277`

This refers to the original version of the casks installed but it doesn't detect that they have been already updated. Any way to fix this?

**Comment by SMillerDev** [comment]
If you can find some way to check the version of what casks install that works across all casks of that type this could work. Unfortunately I've never encountered such a version identifier.

**Reply by cho-m** [reply]
Probably won't be possible across all casks. Would have to be independent to each Cask (e.g. for app bundle, reading `Info.plist` version and transforming it to version format for Cask file kind of like `extract_plist` livecheck). Not sure if worth adding necessary `brew` logic/DSL support for handful of Casks this may be possible on but anyone interested can submit a PR for review.
